Sources/TuistGraph/DependenciesGraph/ThirdPartyDependency.swift

```diff
@@ -2,20 +2,67 @@ import Foundation
 import TSCBasic
 
 // A enum containing information about third party dependency.
-public enum ThirdPartyDependency: Hashable, Equatable, Codable {
+public enum ThirdPartyDependency: Hashable, Codable {
+    /// A dependency that is imported as source code.
+    case sources(products: [Product], targets: [Target]) // TODO: add the supported platforms read from the SPM package to `sources`
```

Comment:
The products are basically the nodes that can be imported from the dependent project.

If I understood correctly, also the other "internal" targets will need to eventually be mapped to tuist targets, because they will be needed to properly compile the products, right?
If yes, we might need to find a way to handle scenarios in which different SPM projects define (internal) target with the same name, which would cause a clash here. In addition to that, they could also clash with user defined targets.

Reply:
> If I understood correctly, also the other "internal" targets will need to eventually be mapped to tuist targets, because they will be needed to properly compile the products, right?

I believe so.

> If yes, we might need to find a way to handle scenarios in which different SPM projects define (internal) target with the same name, which would cause a clash here. In addition to that, they could also clash with user defined targets.

We should also think about how the dependencies will be structured. It could either be:

```
- Dependencies.xcodeproj
---- Sources/RxSwift /Alamofire ...
- MyProject.xcodeproj
...
```

or 
```
- RxSwift.xcodeproj
---- Sources/...
- Alamofire.xcodeproj
---- Sources/...
- MyProject.xcodeproj
...
```

(or something else?)

I am more inclined to the first option as it causes less clutter. 

I think it'd best if you try out one of the above structures by hand, adding targets of the same name to "RxSwift", "Alamofire", and "MyProject" and see how it behaves.

Reply:
Definitely, I have already started doing some experiments and I'll let you know 🙌

Reply:
I have done some tests defining a `Lib1` project and a `Lib2` project, both containing a `Utils` target, and tried to use both as dependencies in another project.
Looks like it doesn't work as they try to generate the same derived data:
```
Multiple commands produce '/Users/df/Library/Developer/Xcode/DerivedData/App-gbuhffkqnrinzlddfzibzmfclseb/Build/Products/Debug-iphonesimulator/libUtils.a':
1) Target 'Lib1Utils' (project 'App'): Libtool /Users/df/Library/Developer/Xcode/DerivedData/App-gbuhffkqnrinzlddfzibzmfclseb/Build/Products/Debug-iphonesimulator/libUtils.a normal
2) Target 'Lib2Utils' (project 'App'): Libtool /Users/df/Library/Developer/Xcode/DerivedData/App-gbuhffkqnrinzlddfzibzmfclseb/Build/Products/Debug-iphonesimulator/libUtils.a normal
```

I have tried to reproduce the same scenario using SPM to handle dependencies (with `Utils` being an internal target not exposed in the products list), and it looks the same is not possible with SPM either:
```swift
swift build                                                                                                                                                                                                                                 
error: multiple targets named 'Utils' in: Lib1, Lib2
error: ExitCode(rawValue: 1)
```

Hence, I think we can ignore this edge case for now 🙌